docs/howto/custom-shell.txt

```diff
@@ -0,0 +1,82 @@
+============================================= 
+How to customize the ``shell`` command 
+=============================================
+
+Django provides functionality to auto-import project models in the shell. In
+your project, you might want to either avoid importing models or include
+additional imports in the shell. You can do this by subclassing the
+:djadmin:`shell` command and overriding a method. To set up the directory of
+your project you can follow this :doc:`guide
+</howto/custom-management-commands>` on how to override an existing command. 
+
+.. note::
+
+    The auto-import feature registers your models looking for
+    :setting:`INSTALLED_APPS` in reverse. So, the models listed first, in
+    :setting:`INSTALLED_APPS` order, will override the models with the same
+    name. In order to access those models that were overridden, you can use the
+    ``<app_label>_models`` module notation, where ``app_label`` is the app whose
+    models you want to access.
+
+Customizing the shell
+=====================
+
+To override the current shell's functionality, you have to override the
+``shell.Command.get_namespace`` in order to return a namespace that maps every
+name to the object imported. For example, if you want to import ``resolve`` and
+``reverse`` methods in you shell, you can edit
+``polls/management/commands/shell.py`` to look like this::
+
+    from django.core.management.commands import shell
+
+
+    class Command(shell.Command):
+        def get_namespace(self):
+            from django.urls.base import resolve, reverse
+
+            return {
+                **super().get_namespace(),
+                "resolve": resolve,
+                "reverse": reverse,
+            }
+
+This way, your customized shell will import models from your apps along with the
+``resolve`` and ``reverse`` methods, allowing you to use them without needing
+additional imports.
+
+.. note::
+
+    If you prefer your shell without any auto-imported model, you can omit
+    calling the ``**super().get_namespace()`` method when returning the
+    namespace, as it can be empty.
+
+
+Implementing your own python shell runner
```

Comment:
I think this is additional in the how to and not related to the auto-importing feature - I would pull this section out to a separate PR